### ISSUE TITLE
[Python] Segfault occurs when subscribing to an intent out of the context manager

### ISSUE THREAD

**Issue by cpoisson** [snipsco/hermes-protocol#39]
**Version**
hermes-python 0.1.27
Python 2 and 3

**Test script**

```python
from hermes_python.hermes import Hermes

def search_weather_callback(hermes, intent_message):
    print("received! {}".format(intent_message.intent.intent_name))

hermes = Hermes("localhost:1883")
hermes.subscribe_intent("SearchWeatherInfo", search_weather_callback)
hermes.loop_forever()
```

=> segfault at subscribe, should throw an exception instead.

**Traces**

```bash
exception, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:102
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:103
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:104
call, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:104
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:104
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:105
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:106
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:107
return, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:107
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:108
line, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:109
return, /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py:109
call, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/hermes.py:53
line, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/hermes.py:54
line, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/hermes.py:58
return, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/hermes.py:58
return, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/hermes.py:60
line, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/hermes.py:77
line, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/hermes.py:78
call, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/ffi/utils.py:21
line, /Users/charles/.virtualenvs/hermes-py2.7/lib/python2.7/site-packages/hermes_python/ffi/utils.py:22
[1]    14265 segmentation fault  python test_hermes.py
```

**Comment by anthonyray** [comment]
This bug comes from the fact that no connection to the MQTT bus was created, because nor the __enter__ and __exit__ methods were called. 
These methods are normally called by the context manager. 

While this was originally a way to restrict the API of the hermes client, and a clean way to achieve graceful exits of action code, we realised that it constrained the API too much and hinders testability and flexibility of the client. 

In the next release of `hermes-python` we will add two convenience methods : `connect` and `disconnect`. These methods will be part of the public API, and `__enter__ ` and `__exit__` will wrap them.

**Comment by anthonyray** [comment]
This is fixed in verison `0.1.28` (https://github.com/snipsco/hermes-protocol/pull/42)

**Comment by koenvervloesem** [comment]
This test script from @cpoisson still segfaults with hermes-python 0.3.3 and Python 3.6.7.
